dynawo/sources/Modeler/ModelManager/DYNModelManager.cpp

```diff
@@ -342,11 +362,24 @@ ModelManager::evalJtAdept(const double t, double* y, double* yp, const double cj
         double term = coeff * jac[indice] + cj * jac[indice + offsetJPrim];
 #ifdef _DEBUG_
         if (isnan(term) || isinf(term)) {
-          throw DYNError(Error::MODELER, JacobianWithNanInf, name(), modelType(), staticId(), i, getFequationByLocalIndex(i), j);   // i is local index
+          throw DYNError(Error::MODELER, JacobianWithNanInf, name(), modelType(), staticId(), i, getFequationByLocalIndex(i), indice);   // i is local index
         }
 #endif
         Jt.addTerm(j + rowOffset, term);
       }
+
+      // assuming that external connections array has been set during initialization if external variables are handled
+      for (unsigned int j = 0; j < sizeYExternal(); j++) {
+        int index = 2 * sizeY() * sizeY() + i + j * sizeF();
+        double term = coeff * jac[index] + cj * jac[index + offsetJPrimExternal];
+        int referenceIndex = getReferenceIndex(j);
+#ifdef _DEBUG_
+        if (isnan(term) || isinf(term)) {
+          throw DYNError(Error::MODELER, JacobianWithNanInf, name(), modelType(), staticId(), i, getFequationByLocalIndex(i), referenceIndex);
```

Comment:
Please create a specific error for ext var with the proper index as the index is the matrix is meaningless for the user

Reply:
OK